### DISCUSSION TITLE
[custom-plugin-lib] Is it possible to customize the logged date format?

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#481] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

v12

### Description

Hi guys,
is it possible to customize the format of the date logged by the custom-plugin-lib?

We don't like so much the epoch (I like it honestly, but not while troubleshooting microservices) because it's not human readable (unless you are so good with math).

Is it possible to customize it or we need to open a Feature Request?
If a feature request is needed, should we open it on `lc39` or  on the `custom-plugin-lib`?

Thanks in advance.

### Actual Outcome

Not human readable date

### Expected Outcome

Human readable date

**Comment by edocrippaofficial** [comment]
The formatter is a property of Pino: in your case the best option is the one specifically for the [timestamp](https://getpino.io/#/docs/api?id=timestamp-boolean-function).

You want to pass to Pino this option:
```
timestamp: pino.stdTimeFunctions.isoTime,
```

The correct place is lc39, specifically [here](https://github.com/mia-platform/lc39/blob/b45585bfb56b7419ead370286bb3e1371bc7838b/lib/custom-logger.js#L37) I think.

**Reply by edocrippaofficial** [reply, marked as answer]
Yep I tried, you can specify this in your `index.js`
```js
module.exports.options = {
  logger: {
    timestamp: () => `,"time":"${new Date(Date.now()).toISOString()}"`,
  },
}
```

In this way `lc39` will use this function as the Pino configuration object.

This is the output:
```
{"level":30,"time":"2024-04-29T19:09:11.002Z","pid":71486,"hostname":"NB004C","msg":"Server listening at http://0.0.0.0:3000"}
```

Maybe your best option is something like:
```js
timestamp: () => `,"time":${Date.now()}, "timeIso":"${new Date(Date.now()).toISOString()}"`,
```

So you can get both versions for automatic tools and for humans 😃

**Reply by frengooo** [reply]
Hi @EdoCrippaOfficial, I think this is the best discussions answer ever 😄 thanks for your time 💪 I'll try it.
